### PR TITLE
Fix storyboard videos not correctly filling the screen

### LINKED ISSUE #12291: Background videos are scaled to fit width and height of the screen resulting in some beatmaps having incorrectly sized video backgrounds.
**Describe the bug:** 
The osu! lazer client scales video to fit the screen's width and height, where the osu! client only resizes video to fit the screen's width.

**Screenshots or videos showing encountered issue:** 
http://prntscr.com/11462jy

**osu!lazer version:** 
2021.331.0

### PR DESCRIPTION
**Pull request by peppy**
Storyboard layers were coded in a way to allow correct positioning of elements, where in stable the coordinate space is 640x480 (even in the case of widescreen). This was limiting elements within `StoryboardLayer`s that have `RelativeSize.Both` specification to that 4:3 aspect ratio, which isn't what we want for videos.

In addition, lazer was missing the special case in stable where a storyboard without any actual storyboard elements (ie. only containing video / backgrounds) was set to `WidescreenStoryboard` implicitly.

Closes https://github.com/ppy/osu/issues/12291.

### CODE REVIEW

File: osu.Game/Storyboards/Drawables/DrawableStoryboard.cs [hunk 1]
```diff
@@ -57,7 +57,13 @@ public bool Passing
         public DrawableStoryboard(Storyboard storyboard)
         {
             Storyboard = storyboard;
+
             Size = new Vector2(640, 480);
+
+            bool onlyHasVideoElements = !Storyboard.Layers.Any(l => l.Elements.Any(e => !(e is StoryboardVideo)));
```
Comment (peppy) [inline osu.Game/Storyboards/Drawables/DrawableStoryboard.cs:63]: Can also be written as

```suggestion
            bool onlyHasVideoElements = !Storyboard.Layers.Where(l => !(l is StoryboardLayerVideo)).Any(l => l.Elements.Any());
```

... but I *think* what I have is easier to mentally parse.

Comment (smoogipoo) [reply osu.Game/Storyboards/Drawables/DrawableStoryboard.cs:63]: Or, `Storyboard.Layers.SelectMany(l => l.Elements).Any(e => !(e is StoryboardVideo))`?

File: osu.Game/Storyboards/StoryboardLayerVideo.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+using osu.Framework.Graphics;
+using osu.Game.Storyboards.Drawables;
+using osuTK;
+
+namespace osu.Game.Storyboards
+{
+    public class StoryboardLayerVideo : StoryboardLayer
+    {
+        public StoryboardLayerVideo(string name, int depth, bool masking)
```
Comment (smoogipoo) [inline osu.Game/Storyboards/StoryboardLayerVideo.cs:12]: Naming feels backwards. `StoryboardVideoLayer`?

File: osu.Game/Storyboards/Drawables/DrawableStoryboard.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Storyboards/Drawables/DrawableStoryboard.cs:63]: Sure, that works as well. Not sure which is more efficient if any, but I guess it doesn't matter too much in this usage.

File: osu.Game/Storyboards/StoryboardLayerVideo.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Storyboards/StoryboardLayerVideo.cs:12]: That's how I had it, then i double-took. Will change back.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**